### ISSUE TITLE
onnxruntime with CUDA not releasing about 400 MB memory after the session and environment is destroyed

### ISSUE THREAD

**Issue by suneesh** [microsoft/onnxruntime#14819]
### Describe the issue

After creating an onnxruntime session with CUDA, overall CPU memory raises and doesn't go back to the previous state after the release of session. There is always a 400 MB memory residue.

### To reproduce

Check memory before and after running the code snippet
```
// Ort::ENV creation
  Ort::Env OrtEnv(ORT_LOGGING_LEVEL_ERROR, "ORT initialization ");

  // Ort::SessionOptions creation
  Ort::SessionOptions session_options;
  OrtSessionOptionsAppendExecutionProvider_CUDA(session_options, 0);
    
   Ort::Session OrtSession = Ort::Session(OrtEnv, _T("AutoDV.onnx"), session_options);

```

### Urgency

Yes, our product runs on a memory constrained enviornment. This is a blocker

### Platform

Windows

### OS Version

10

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

)onnxruntime-win-x64-gpu-1.12.1

### ONNX Runtime API

C++

### Architecture

X64

### Execution Provider

CUDA

### Execution Provider Library Version

Cuda 11.6

**Comment by pauldog** [comment]
Hello, is this the same issue as: #14466 and #14815 ??

**Comment by pranavsharma** [comment]
Are you sure the session is getting destructed?

**Comment by suneesh** [comment]
> 

Yes, I debugged and the destructor for Env and Session is invoked. 
I could see the CUDA provider is also destroyed.

**Comment by suneesh** [comment]
> Hello, is this the same issue as: #14466 and #14815 ??
Could be due to same underlying cause.

**Comment by pranavsharma** [comment]
@suneesh - we've fixed a leak in [this PR](https://github.com/microsoft/onnxruntime/pull/15040). It relates to using external weights. Are you using external weights too? Can you provide a repro?
